providers/azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/VirtualNetworkApiMockTest.java

```diff
@@ -53,6 +56,36 @@ public void getVirtualNetwork() throws InterruptedException {
       assertEquals(vn.tags().get("tagkey"), "tagvalue");
    }
 
+   public void checkIpAvailable() throws InterruptedException {
+      server.enqueue(jsonResponse("/ipavailable.json"));
+
+      final VirtualNetworkApi vnApi = api.getVirtualNetworkApi(resourcegroup);
+      IpAddressAvailabilityResult checkResult = vnApi.checkIPAddressAvailability(virtualNetwork, ipAvailable);
+
+      String path = String.format("/subscriptions/%s/resourcegroups/%s/providers/Microsoft"
+                  + ".Network/virtualNetworks/%s/CheckIPAddressAvailability?ipAddress=%s&%s", subscriptionid,
+            resourcegroup,
+            virtualNetwork, ipAvailable, apiVersion);
+      assertSent(server, "GET", path);
+      assertTrue(checkResult.available());
+      assertTrue(checkResult.availableIPAddresses().isEmpty());
+   }
+
+   public void checkIpNotAvailable() throws InterruptedException {
```

Comment:
This test is redundant. In Mock tests we just want to assert that given an input we generate the right request. Having verified the request, and that the response can be deserialized, the rest is out of the scope of the test, so let's just remove this one.

Reply:
The point is that the response can change (has an extra field). Maybe the redundant one is the checkIpAvailable since that one only has one field?

Reply:
One of the two methods is redundant. Keep the one you prefer